internal/server/server.go

```diff
@@ -416,8 +413,42 @@ func (s *Server) handleHealthCheck() http.HandlerFunc {
 				"err":        err,
 			}).Error(errPrefix + status)
 
-			writeJsonResponse(w, http.StatusBadGateway, healthCheckResponse{Status: status})
-			return
+			// Code below is for debugging the /healthz hanging issue
+			// Once issue is resolved, this block can be deleted
+			// Debug Start
+
+			// try another service instead of OIDC provider
+			resp, err := s.client.Get(s.debugURL.String())
+			if err == nil {
+				log.Info(errPrefix + fmt.Sprintf("s.client.Get worked for: %v", s.debugURL))
+				resp.Body.Close()
+			} else {
+				log.Info(errPrefix + fmt.Sprintf("s.client.Get also failed for: %v", s.debugURL))
+			}
+
+			// retry with a new HTTP Client
+			if err := s.initHttpClient(); err != nil {
```

Comment:
For the sake of clarity, let's add a log line before this that explicitly states the shared http client is being replaced.

Reply:
Fixed